### PR TITLE
[DERCBOT-805] IA Orchestrator Client & Validation RAG

### PR DESCRIPTION
**Pull request by assouktim**

### CODE REVIEW

**Comment by assouktim** [comment]
/!\ Do not merge on rag/dercbot-768 /!\

**Review by Benvii — CHANGES_REQUESTED** [review]
- [x]  Naming AIOrchestrator may be confusing as we have AI in NLP API, maybe genAiOrchestrator is best suited
- [x] Stick to snake_case for the API but configure retro fit to do the conversion

File: llm/orchestrator-core/src/main/kotlin/ai/tock/llm/orchestrator/core/models/em/EMSetting.kt [hunk 1]
```diff
@@ -27,11 +27,10 @@ import com.fasterxml.jackson.annotation.JsonTypeInfo
     property = "provider"
 )
 @JsonSubTypes(
-    JsonSubTypes.Type(value = OpenAIEMSetting::class, name = LLMConstants.OPEN_AI),
-    JsonSubTypes.Type(value = AzureOpenAIEMSetting::class, name = LLMConstants.AZURE_OPEN_AI_SERVICE)
+    JsonSubTypes.Type(value = OpenAIEMSetting::class, name = Constants.OPEN_AI),
+    JsonSubTypes.Type(value = AzureOpenAIEMSetting::class, name = Constants.AZURE_OPEN_AI_SERVICE)
 )
 abstract class EMSetting(
     val provider: LLMProvider,
```
[the remark this answers is not in the document]
Comment (assouktim) [reply llm/orchestrator-core/src/main/kotlin/ai/tock/llm/orchestrator/core/models/em/EMSetting.kt:34]: ah yes, I had started to do it, I created the missing enum, but I didn't use it

**Review by Benvii — APPROVED** [review]
LGTM

**Review by Benvii — CHANGES_REQUESTED** [review]
Camel case VS Snake Case still need to be done in retrofit.

File: llm/orchestrator-core/src/main/kotlin/ai/tock/llm/orchestrator/core/models/em/EMSetting.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Benvii) [reply llm/orchestrator-core/src/main/kotlin/ai/tock/llm/orchestrator/core/models/em/EMSetting.kt]: ok

File: llm/orchestrator-server/src/main/python/app/src/llm_orchestrator/models/em/em_setting.py [hunk 2]
```diff
@@ -22,4 +22,5 @@ class BaseEMSetting(BaseModel):
     api_key: str = Field(
         description='The API key used to authenticate requests to the provider API.',
         examples=['123-abc-456-def'],
+        alias="apiKey"
```
[the remark this answers is not in the document]
Comment (assouktim) [reply llm/orchestrator-server/src/main/python/app/src/llm_orchestrator/models/em/em_setting.py]: WIP

[the remark this answers is not in the document]
Comment (assouktim) [reply llm/orchestrator-server/src/main/python/app/src/llm_orchestrator/models/em/em_setting.py]: it's done

**Review by Benvii — APPROVED** [review]
Thanks it's ok

**Pull request merged**